pravega-k8-log-collection/pravega-k8s-log.sh

```diff
@@ -0,0 +1,34 @@
+#/usr/bin/env bash
+set -euo pipefail
+mkdir -p /tmp/logdir
+output_dir="/tmp/logdir"
+kubectl get po,svc,pvc --all-namespaces > $output_dir/kubectl-get-po-svc-pvc-all-namespace_output
+IFS=$'\n'
+for i in $(cat $output_dir/kubectl-get-po-svc-pvc-all-namespace_output | grep -i pravega)
+do
+echo $i
+if [[ $i == *"po/"* ]] || [[ $i == *"pod/"* ]]; then
+    if [ "`echo $i | awk '{print $4}'`" == "Running" ]; then
+    {
+        if [ `echo $i | awk '{print $5}'` != 0 ]; then     
+            kubectl -n `echo $i | awk '{print $1}'` logs `echo $i | awk '{print $2}'` -p > $output_dir/log_`echo $i | awk 'gsub(/\//,"_") {print $2}'`_previous.log
+        fi
+        kubectl -n `echo $i | awk '{print $1}'` logs `echo $i | awk '{print $2}'` > $output_dir/log_`echo $i | awk 'gsub(/\//,"_") {print $2}'`_recent.log
+        log_count=$(kubectl -n `echo $i | awk '{print $1}'` exec -it `echo $i |  awk '-F'[/] '{print $2}'|awk '{print $1}'` -- ls -l /opt/pravega/logs 2>/dev/null | wc -l || true)
+        if [ $log_count -ge 2 ]; then  
```

Comment:
I'm not sure what this if block is doing. Why are these `oldlogs`?

Reply:
This is because of current design of `kubectl logs`. There was discussion in k8 spaces: https://github.com/kubernetes/kubernetes/issues/11046, on how much logging it would produce and decision was "I think we should advise that kubectl logs is just a "cache" of some of the logs for a pod"

So, with that scope, `kubectl logs` will not collect complete log from the pods in occasions, especially when logs are increasing. Our requirement is, we need to collect complete logs for the pods for a customer deployment  for debugging. Currently logs get rotate every day or if it reach 10MB and `kubectl logs` points to recent logging. See below example:

```
root@osboxes:~# kubectl -n nautilus-pravega exec -it nautilus-segmentstore-2  -- ls -l /opt/pravega/logs
total 3340
-rw-r--r--    1 root     root       2844579 Dec 16 16:21 segmentstore.log
-rw-r--r--    1 root     root        571209 Dec 16 15:37 segmentstore_2018-12-16.0.log.gz
```
Here log rotate happened and recent log is pointing to following:
```
2018-12-16 16:26:18,135 37002858 [core-17] INFO  i.p.s.s.h.ZKSegmentContainerMonitor - Container Changes: Desired = [0, 3], Current = [0, 3], PendingTasks = [], ToStart = [], ToStop = [].
2018-12-16 16:26:28,136 37012859 [core-12] INFO  i.p.s.s.h.ZKSegmentContainerMonitor - Container Changes: Desired = [0, 3], Current = [0, 3], PendingTasks = [], ToStart = [], ToStop = [].
2018-12-16 16:26:38,136 37022859 [core-15]
```
Actual log started 
```
Waiting for nautilus-pravega-zookeeper-client:2181
Connected to nautilus-pravega-zookeeper-client:2181
SLF4J: Found binding in [jar:file:/opt/pravega/lib/logback-classic-1.2.3.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/opt/pravega/lib/slf4j-log4j12-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
06:09:35,298 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [/opt/pravega/conf/logback.xml] at [file:/opt/pravega/conf/logback.xml]
06:09:35,434 |-INFO in ch.qos.logback.classic.jo
```
So what this block is doing, it is checking if there is any log files present (rotated or non-rotated form) and if there are files take the complete directory archive of that moment, which will collect all rotated logs and current log completely.